### ISSUE TITLE
Some types of bushes appear to break game conventions for item piles in ASCII (0.C-3076 cac8af7)

### ISSUE THREAD

**Issue by graysage1** [CleverRaven/Cataclysm-DDA#12202]
It used to be that ASCII tiles with solid backgrounds identified item piles.

Now there are some bushes that have solid backgrounds all on their own, even if there is no item pile in that bush.  One of the offending bushes is 'strawberries', but there may be others.

It's important to stick to conventions if we have them.  If solid background means there is an item pile, and that's our convention, we should try to stick to it consistently.
##

**Comment by Coolthulhu** [comment]
Bushes with backgrounds do contain items, just not items that are on map.

**Comment by graysage1** [comment]
@Coolthulhu that still breaks conventions, because other trees and bushes that "have items" don't have solid backgrounds.

**Comment by Coolthulhu** [comment]
@graysage1 Other bushes don't have guaranteed items and hide items (underbrush), while flowers and farmed plants essentially are the items themselves.
What's left is fruit trees - they don't have a "common convention" with fruit bushes other than just plain visibility (old fruit bushes did blend in too much).

**Comment by graysage1** [comment]
@Coolthulhu 

>What's left is fruit trees

And pines.  Pines have guaranteed drops, but don't have a solid background.  I still think there is a big difference between a guaranteed drop, which is something you know to expect, and hidden terrain-unrelated items in the item pile.

It's much more important to tell the player that there is a pile than something we know to expect, like plant drops.

I do agree that plants with plant drops should be distinctive, but I'd like it better if solid backgrounds were still reserved for item piles as in the past.

**Comment by Coolthulhu** [comment]
@graysage1 If you have a better solution, it would be good to know. Different foreground colors for bushes didn't work (we tried - no good colors). The solution is also better than the problem was, since edible plants are far more important than items you'd expect to find near them (and rare exceptions can be solved with 'V').

It has to fulfill the following requirements:
* Can't be too similar to something else (wreckage, dirt pile, shrub, underbrush)
* Has to be visible enough
* Has to have "reasonable" glyph

The only other solution for bushes I can think of would be making them '$' glyph, but then they'd have to only use colors like green, light green and yellow to blend in.
Breaking the former convention to establish a new one would be beneficial here.

If you're so concerned about fruit and items sharing tiles, fruit shrubs could get a NOITEMS tag, like farmed plants got.

**Comment by graysage1** [comment]
@Coolthulhu 

>Different foreground colors for bushes didn't work (we tried - no good colors).

I hear you, OK.

How about this then: let's reserve solid green background for bushes with drops and leave every other solid background for item piles?  Would that work?

**Comment by Coolthulhu** [comment]
Solid colors are used for:
* Vehicle tiles with items (foreground and background get swapped)
* Status effects on monsters (NPCs don't get this one yet, but will)
* Items (only solid blue is used here)
* Cosmetic fields under items

**Comment by graysage1** [comment]
@Coolthulhu 

>Items (only solid blue is used here

I'm pretty sure I've been seeing some item piles with non-blue backgrounds.

**Comment by Coolthulhu** [comment]
Ones with cosmetic fields underneath them, yes.

**Comment by graysage1** [comment]
So are we giving up on conventions then?  It seems you're arguing against conventions by citing all these exceptions, as if they were good things.

**Comment by Coolthulhu** [comment]
I'm citing examples of solid background usage in the game to show you that there was no "only items can have solid background" convention and that it would be a rather unreasonable one to start enforcing right now.

Conventions are there for a reason. If the reason to keep a convention is weaker than the reason to break the current one and get a less strict one, then it should be broken. In this case visibility is better than all plants only using the glyph and foreground color.

Few more examples:
* Zeds
* Some terrain (gas station tanks, fermenting vats)

DDA could use some more solid backgrounds. For example, smoke having it instead of covering everything underneath, especially fire.

**Comment by graysage1** [comment]
Which tiles are considered 'cosmetic'?  I'm wondering how important it is to give cosmetic tiles solid backgrounds.

>Status effects on monsters (NPCs don't get this one yet, but will)

This doesn't introduce confusion because I know I just caused it by hitting the monster.  It's not like I am seeing a far-off tile that I haven't interacted with and am wondering what's there.  So this case is defensible.

>For example, smoke having it instead of covering everything underneath, especially fire.

Smoke is meant to obscure vision, no?  Smoke even impedes line of sight the way a solid wall tile would.  So smoke hiding fire seems like a legitimate hazard that should be kept as a hazard.  If you wade though smoke and step on a fire, then too bad.  Putting infrared goggles on and activating them should show the fires, however, even through smoke.  I don't know if it already works like that.  But I don't think any of this requires the use of solid backgrounds.

I agree that sometimes it's good to break conventions, but not until every other option has been considered and exhausted.

**Comment by i2amroy** [comment]
The big problem with smoke isn't so much that heavy smoke conceals things, it's that even the faintest of smoke totally obscures whatever is beneath it. As a result something like lighting a campfire can totally obscure what the terrain is for quite a large area.

**Comment by graysage1** [comment]
Firstly, not all item piles are blue.

For example, on my base I have a pile like this:

```
a - heavy fletched fire-hardened arrow (40)
b - Computer Gaming
```

And its background color is magenta I believe.  It's definitely not blue.  And I've seen other such things.  This item pile is on the floor, not on some special tile.  I just moved the pile over (and discovered a bug in AIM, I'll open another issue for this), and now Computer Gaming is on top.  As a result, the item pile has a green background on a plain floor.

So firstly, let's agree that items piles are no longer limited to blue.  Maybe they should go back to being limited to blue, but currently that's not the case.

@i2amroy There are ways of dealing with all this without making things ambiguous.

What if all items piles had blue background, bushes with berries were green background, and smoke had gray?  Then there wouldn't be confusion anymore.  But this would require reserving various background colors for various uses.

**Comment by Coolthulhu** [comment]
> Smoke is meant to obscure vision, no?

Survivor vision, not player vision.

> But I don't think any of this requires the use of solid backgrounds.

Doesn't need to need, "would heavily benefit from" is quite good.

> Which tiles are considered 'cosmetic'? I'm wondering how important it is to give cosmetic tiles solid backgrounds.

Blood, guts, slime etc. Anything that can't hurt you.

> And its background color is magenta I believe. It's definitely not blue.

My mistake.
Now I remember: items (like vehicles) swap bg/fg when multiple items are on the tile.
Making them all just have blue background would make sense, though. It would cut down on visual rainbow. Well, except for blue items with more items - that would need some good solution.

> What if all items piles had blue background, bushes with berries were green background, and smoke had gray? Then there wouldn't be confusion anymore.

Would be fine by me.

**Comment by graysage1** [comment]
>except for blue items with more items

You're talking about tiles with blue foreground?  Hmm... Maybe item piles will need to use two background colors then.  Blue and some other one in case the foreground of a tile is also blue.

Or maybe item piles should use a color that looks nice as a background but looks poorly as foreground, like say magenta, and then never use that color for the foreground of anything.

What happens if smoke covers a light gray foreground tile? :)  Is smoke going to be multihued?

In any case, it appears that right now item piles use many different backgrounds without any pressing need for it.  I suppose the background is supposed to be influenced by what's on top of the pile, but it's not informative.  I still have to look closely inside the pile to know what's there.

**Comment by Coolthulhu** [comment]
Picking an arbitrary color for same color collision could also work.
Say, blue fg + blue bg gives black fg + blue bg.
Same for smoke.

There are some rare effects that set foreground color (light smoke, like weed or meth). Those could be moved to background too.

**Comment by graysage1** [comment]
@Coolthulhu Good idea.  I overlooked that you can also tinker with the fg color if necessary.

**Comment by graysage1** [comment]
Not that it really matters, but it looks like the background color of an item pile is controlled by the bottommost item, not the topmost.

**Comment by kevingranade** [comment]
It means there are berries there.  I'd say it's extending the convention
rather than breaking it.

**Comment by graysage1** [comment]
Alright so most people think this is a non-issue.  Not a problem.  I should probably close this one.

**Comment by Coolthulhu** [comment]
Leave it open for now.

**Comment by Zireael07** [comment]
+1 to making trees look like bushes.
